# Examen de l'export CSV du logiciel @ctes

Rapide examen de l'export CSV d' @actes.


## TODO

* [ ] bug report pandera: il faut échapper les single quotes dans les intitulés de colonne (ex: "Let's see") : `schema_export = pa.infer_schema(df_export.rename(columns={x: x.replace("'", "\\'") for x in df_export.columns}))`

## Chargement des données

Commençons par charger les données dans pandas.

In [1]:
from pathlib import Path

import pandas as pd
import pandera as pa

from pandera import Check, Column, DataFrameSchema

In [2]:
CSV_RECHER = Path("../data/raw/recherche-actes-20220308-1126.csv")
CSV_EXPORT = Path("../data/raw/export-actes-20220308-1126.csv")

In [3]:
# le début des fichiers contient 12 lignes non-vides de métadonnées de recherche
# les intitulés de colonnes sont donc sur la ligne 13 (index 12)
dtype_actes = {
    "N° Acte": "string",
    "Nature": "category",
    "Matière": "category",
    "Objet": "string",
    "État de l'acte": "category",
    "Caractère prioritaire": "category",
    "Mode de transmission": "category",
    "Acte contrôlé": "category",
    "Acte annulé": "category",
    "Service attributaire": "category",
    "Date de transmission du courrier simple": "string",  # TODO date?
    "Date de transmission de la lettre d'observation": "string",  # TODO date?
    "Date de demande de transmission complémentaire": "string",  # TODO date?
    "Date de saisie de l'expert": "string",  # TODO date?
    "Date Réponse Expert": "string",  # TODO date?
    "Nature réponse Expert": "string",  # TODO category? absence de valeurs
    "Libellé émetteur (ACTES)": "category",  # numéro d'arrondissement puis commune
    "Nature émetteur": "category",
    "Tiers de transmission": "category",
    "Département": "category",
    "Arrondissement": "category",
    "Région": "category",
    "Strate": "category",
}

df_recher = pd.read_csv(CSV_RECHER, encoding="iso-8859-15", sep=';', header=12, dtype=dtype_actes)
df_export = pd.read_csv(CSV_EXPORT, encoding="iso-8859-15", sep=';', header=12, dtype=dtype_actes)

In [4]:
df_recher

N° Acte                Nature  \
0         013-211300058-20220307-070322_01-AR  Actes réglementaires   
1    013-211300553-20220304-2022_00515_VDM-AR  Actes réglementaires   
2               013-211300280-20220302-174-AR  Actes réglementaires   
3    013-211300553-20220301-2022_00526_VDM-AR  Actes réglementaires   
4    013-211300553-20220301-2022_00535_VDM-AR  Actes réglementaires   
..                                        ...                   ...   
109         013-211300470-20220107-52-2022-AR  Actes réglementaires   
110         013-211300470-20220105-05-2022-AR  Actes réglementaires   
111  013-211300553-20220104-2021_04230_VDM-AR  Actes réglementaires   
112  013-211300553-20211231-2021_04228_VDM-AR  Actes réglementaires   
113  013-211300553-20211231-2021_04227_VDM-AR  Actes réglementaires   

                               Matière  \
0              6.1 - Police municipale   
1              6.1 - Police municipale   
2              6.1 - Police municipale   
3              6.1 - Police municipale   
4              6.1 - Police municipale   
..                                 ...   
109            6.1 - Police municipale   
110  6.4 - Autres actes reglementaires   
111            6.1 - Police municipale   
112            6.1 - Police municipale   
113            6.1 - Police municipale   

                                                 Objet État de l'acte  \
0    ARRETE MISE EN SECURITE PROCEDURE ORDINAIRE LE...     A examiner   
1    Arrêté municipal autorisant la mise en ?uvre d...     A examiner   
2    Mise en sécurité de travaux d'urgence dans l'i...     A examiner   
3    sdi 20/047 - arrêté modificatif de mise en séc...     A examiner   
4    SDI 22/037 - Arrêté de Mise en Sécurité - Proc...     A examiner   
..                                                 ...            ...   
109  Mise en demeure de faire pratiquer une évaluat...     A examiner   
110  Mise à jour du règlement intérieur du château ...     A examiner   
111  SDI 21/820 - ARRÊTÉ DE MISE EN SÉCURITÉ - PROC...     A examiner   
112  SDI 19/009- ARRÊTÉ MODIFICATIF DE MISE EN SECU...     A examiner   
113  sdi 21/360 - arrêté de mise en sécurité - 2 ru...     A examiner   

    Caractère prioritaire Mode de transmission Acte contrôlé Acte annulé  \
0                     NaN         Télétransmis           NaN         NaN   
1                     NaN         Télétransmis           NaN         NaN   
2                     NaN         Télétransmis           NaN         NaN   
3                     NaN         Télétransmis           NaN         NaN   
4                     NaN         Télétransmis           NaN         NaN   
..                    ...                  ...           ...         ...   
109                   NaN         Télétransmis           NaN         NaN   
110                   NaN         Télétransmis           NaN         NaN   
111                   NaN         Télétransmis             1         NaN   
112                   NaN         Télétransmis             1         NaN   
113                   NaN         Télétransmis             1         NaN   

                                  Service attributaire  ...  \
0    Domaine départemental - CL - Préfecture des Bo...  ...   
1    Domaine départemental - CL - Préfecture des Bo...  ...   
2    Domaine départemental - CL - Préfecture des Bo...  ...   
3    Domaine départemental - CL - Préfecture des Bo...  ...   
4    Domaine départemental - CL - Préfecture des Bo...  ...   
..                                                 ...  ...   
109  Domaine départemental - CL - Préfecture des Bo...  ...   
110  Domaine départemental - CL - Préfecture des Bo...  ...   
111  Domaine départemental - CL - Préfecture des Bo...  ...   
112  Domaine départemental - CL - Préfecture des Bo...  ...   
113  Domaine départemental - CL - Préfecture des Bo...  ...   

    Date de saisie de l'expert Date Réponse Expert Nature réponse Expert  \
0                         <NA>                <NA>       

In [5]:
df_export

N° Acte                Nature  \
0        013-211300058-20220307-070322_01-AR  Actes réglementaires   
1              013-211300280-20220302-174-AR  Actes réglementaires   
2   013-211300553-20220301-2022_00526_VDM-AR  Actes réglementaires   
3   013-211300553-20220301-2022_00535_VDM-AR  Actes réglementaires   
4   013-211300553-20220301-2022_00527_VDM-AR  Actes réglementaires   
..                                       ...                   ...   
95  013-211300553-20220106-2022_00018_VDM-AR  Actes réglementaires   
96      013-211300074-20220106-AR-2022-02-AR  Actes réglementaires   
97  013-211300553-20220104-2021_04230_VDM-AR  Actes réglementaires   
98  013-211300553-20211231-2021_04228_VDM-AR  Actes réglementaires   
99  013-211300553-20211231-2021_04227_VDM-AR  Actes réglementaires   

                              Matière  \
0             6.1 - Police municipale   
1             6.1 - Police municipale   
2             6.1 - Police municipale   
3             6.1 - Police municipale   
4             6.1 - Police municipale   
..                                ...   
95            6.1 - Police municipale   
96  6.4 - Autres actes reglementaires   
97            6.1 - Police municipale   
98            6.1 - Police municipale   
99            6.1 - Police municipale   

                                                Objet État de l'acte  \
0   ARRETE MISE EN SECURITE PROCEDURE ORDINAIRE LE...     A examiner   
1   Mise en sécurité de travaux d'urgence dans l'i...     A examiner   
2   sdi 20/047 - arrêté modificatif de mise en séc...     A examiner   
3   SDI 22/037 - Arrêté de Mise en Sécurité - Proc...     A examiner   
4   SDI 19/031 - ARRÊTÉ MODIFICATIF DE MISE EN SÉC...     A examiner   
..                                                ...            ...   
95  SDI 20/110- ARRÊTÉ MODIFICATIF DE MISE EN SECU...     A examiner   
96  ARRETE MODIFICATIF N.2022-02 DE MISE EN SECURI...     A examiner   
97  SDI 21/820 - ARRÊTÉ DE MISE EN SÉCURITÉ - PROC...     A examiner   
98  SDI 19/009- ARRÊTÉ MODIFICATIF DE MISE EN SECU...     A examiner   
99  sdi 21/360 - arrêté de mise en sécurité - 2 ru...     A examiner   

   Caractère prioritaire Mode de transmission Acte contrôlé Acte annulé  \
0                    NaN         Télétransmis           NaN         NaN   
1                    NaN         Télétransmis           NaN         NaN   
2                    NaN         Télétransmis           NaN         NaN   
3                    NaN         Télétransmis           NaN         NaN   
4                    NaN         Télétransmis           NaN         NaN   
..                   ...                  ...           ...         ...   
95                   NaN         Télétransmis             1         NaN   
96                   NaN         Télétransmis           NaN         NaN   
97                   NaN         Télétransmis             1         NaN   
98                   NaN         Télétransmis             1         NaN   
99                   NaN         Télétransmis             1         NaN   

                                 Service attributaire  ...  \
0   Domaine départemental - CL - Préfecture des Bo...  ...   
1   Domaine départemental - CL - Préfecture des Bo...  ...   
2   Domaine départemental - CL - Préfecture des Bo...  ...   
3   Domaine départemental - CL - Préfecture des Bo...  ...   
4   Domaine départemental - CL - Préfecture des Bo...  ...   
..                                                ...  ...   
95  Domaine départemental - CL - Préfecture des Bo...  ...   
96  Domaine départemental - CL - Préfecture des Bo...  ...   
97  Domaine départemental - CL - Préfecture des Bo...  ...   
98  Domaine départemental - CL - Préfecture des Bo...  ...   
99  Domaine départemental - CL - Préfecture des Bo...  ...   

   Date de saisie de l'expert Date Réponse Expert Nature réponse Expert  \
0                        <NA>                <NA>                  <NA>   
1                        <NA>             

## Tentative de définition d'un schéma de données et validation du contenu du fichier

Nous utilisons pandera.

In [6]:
DFSCHEMA_RECHER = Path("../scripts/dfschema_actes_recher.py")
if not DFSCHEMA_RECHER.exists():
    # à exécuter une fois seulement, puis on itère sur le script généré dans scripts/
    schema_recher = pa.infer_schema(df_recher.rename(columns={x: x.replace("'", "\\'") for x in df_recher.columns}))
    schema_recher.to_script(DFSCHEMA_RECHER)

In [7]:
DFSCHEMA_EXPORT = Path("../scripts/dfschema_actes_export.py")
if not DFSCHEMA_EXPORT.exists():
    # à exécuter une fois seulement, puis on itère sur le script généré dans scripts/
    schema_export = pa.infer_schema(df_export.rename(columns={x: x.replace("'", "\\'") for x in df_export.columns}))
    schema_export.to_script(DFSCHEMA_EXPORT)

## Comparaison des deux listes


In [8]:
df_comp = pd.merge(df_recher, df_export, how="outer", indicator=True)
df_comp

N° Acte                Nature  \
0         013-211300058-20220307-070322_01-AR  Actes réglementaires   
1    013-211300553-20220304-2022_00515_VDM-AR  Actes réglementaires   
2               013-211300280-20220302-174-AR  Actes réglementaires   
3    013-211300553-20220301-2022_00526_VDM-AR  Actes réglementaires   
4    013-211300553-20220301-2022_00535_VDM-AR  Actes réglementaires   
..                                        ...                   ...   
109         013-211300470-20220107-52-2022-AR  Actes réglementaires   
110         013-211300470-20220105-05-2022-AR  Actes réglementaires   
111  013-211300553-20220104-2021_04230_VDM-AR  Actes réglementaires   
112  013-211300553-20211231-2021_04228_VDM-AR  Actes réglementaires   
113  013-211300553-20211231-2021_04227_VDM-AR  Actes réglementaires   

                               Matière  \
0              6.1 - Police municipale   
1              6.1 - Police municipale   
2              6.1 - Police municipale   
3              6.1 - Police municipale   
4              6.1 - Police municipale   
..                                 ...   
109            6.1 - Police municipale   
110  6.4 - Autres actes reglementaires   
111            6.1 - Police municipale   
112            6.1 - Police municipale   
113            6.1 - Police municipale   

                                                 Objet État de l'acte  \
0    ARRETE MISE EN SECURITE PROCEDURE ORDINAIRE LE...     A examiner   
1    Arrêté municipal autorisant la mise en ?uvre d...     A examiner   
2    Mise en sécurité de travaux d'urgence dans l'i...     A examiner   
3    sdi 20/047 - arrêté modificatif de mise en séc...     A examiner   
4    SDI 22/037 - Arrêté de Mise en Sécurité - Proc...     A examiner   
..                                                 ...            ...   
109  Mise en demeure de faire pratiquer une évaluat...     A examiner   
110  Mise à jour du règlement intérieur du château ...     A examiner   
111  SDI 21/820 - ARRÊTÉ DE MISE EN SÉCURITÉ - PROC...     A examiner   
112  SDI 19/009- ARRÊTÉ MODIFICATIF DE MISE EN SECU...     A examiner   
113  sdi 21/360 - arrêté de mise en sécurité - 2 ru...     A examiner   

    Caractère prioritaire Mode de transmission Acte contrôlé Acte annulé  \
0                     NaN         Télétransmis           NaN         NaN   
1                     NaN         Télétransmis           NaN         NaN   
2                     NaN         Télétransmis           NaN         NaN   
3                     NaN         Télétransmis           NaN         NaN   
4                     NaN         Télétransmis           NaN         NaN   
..                    ...                  ...           ...         ...   
109                   NaN         Télétransmis           NaN         NaN   
110                   NaN         Télétransmis           NaN         NaN   
111                   NaN         Télétransmis             1         NaN   
112                   NaN         Télétransmis             1         NaN   
113                   NaN         Télétransmis             1         NaN   

                                  Service attributaire  ...  \
0    Domaine départemental - CL - Préfecture des Bo...  ...   
1    Domaine départemental - CL - Préfecture des Bo...  ...   
2    Domaine départemental - CL - Préfecture des Bo...  ...   
3    Domaine départemental - CL - Préfecture des Bo...  ...   
4    Domaine départemental - CL - Préfecture des Bo...  ...   
..                                                 ...  ...   
109  Domaine départemental - CL - Préfecture des Bo...  ...   
110  Domaine départemental - CL - Préfecture des Bo...  ...   
111  Domaine départemental - CL - Préfecture des Bo...  ...   
112  Domaine départemental - CL - Préfecture des Bo...  ...   
113  Domaine départemental - CL - Préfecture des Bo...  ...   

    Date Réponse Expert Nature réponse Expert Libellé émetteur (ACTES)  \
0                  <NA>                  <NA>       3 AUBAG

In [9]:
# on vérifie que l'export est bien un sous-ensemble des résultats de la recherche
assert df_comp[df_comp["_merge"] == "right_only"].empty

In [10]:
# on examine les résultats jugés non pertinents par la DDTM
df_comp[df_comp["_merge"] == "left_only"][["Objet"]]

Objet
1    Arrêté municipal autorisant la mise en ?uvre d...
11   REGLEMENTATION GENERALE   Autorisation d occup...
12   Mise en demeure de faire pratiquer un suivi sa...
13   Mise en demeure de faire pratiquer un suivi sa...
18   Arrêté de prolongation d'autorisation de mise ...
36   OCCUPATION DU DOMAINE PUBLIC   REGLEMENTATION ...
37   Interdiction de remise de fonds, de valeurs ou...
65   Arrete portant autorisation d occupation du do...
80   SDI 22/029 - ARRÊTÉ PORTANT SUR LA MISE EN PLA...
102  Mise en service du guichet unique numérique ur...
103                   mise en demeure enlèvement dépôt
104   mise en demeure evaluation comportementale chien
109  Mise en demeure de faire pratiquer une évaluat...
110  Mise à jour du règlement intérieur du château ...

L'entrée 80 avait été jugée non pertinente bien que son objet soit connexe.

In [20]:
with pd.option_context("display.max_colwidth", 0):
    display(df_comp.iloc[[80]])

N° Acte                Nature  \
80  013-211300553-20220127-2022_00279_VDM-AR  Actes réglementaires   

                    Matière  \
80  6.1 - Police municipale   

                                                                                                                                                     Objet  \
80  SDI 22/029 - ARRÊTÉ PORTANT SUR LA MISE EN PLACE D'UN PÉRIMÈTRE DE SÉCURITÉ - 548 BOULEVARD MIREILLE LAUZE - 13011 MARSEILLE - PARCELLE N°211866 L0199   

   État de l'acte Caractère prioritaire Mode de transmission Acte contrôlé  \
80  A examiner     NaN                   Télétransmis         1              

   Acte annulé                                          Service attributaire  \
80  NaN         Domaine départemental - CL - Préfecture des Bouches du Rhône   

    ... Date Réponse Expert Nature réponse Expert Libellé émetteur (ACTES)  \
80  ...  <NA>                <NA>                  3 MARSEILLE  MAIRIE       

                      Nature émetteur Tiers de transmission Département  \
80  3-1 - Commune ou commune nouvelle  DOCAPOSTE  FAST       13           

   Arrondissement                      Région  \
80  3              Provence-Alpes-Côte-d'Azur   

                                                   Strate     _merge  
80  Population supérieure ou égale  à 1 000 000 habitants  left_only  

[1 rows x 24 columns]

Après vérification par Florence auprès de Marc, les arrêtés de périmètre de sécurité seront à inclure dans le périmètre de l'outil.

## Analyse des résultats exportés

Nous pouvons maintenant nous concentrer sur les seuls résultats exportés, et relever les données exposées dans le champ "Objet" selon l'émetteur.

In [13]:
# laisser le tableau s'étendre en largeur autant que nécessaire, ou autant que possible en faisant du text wrap dans la cellule
with pd.option_context("display.max_colwidth", 0):
    # afficher l'émetteur et l'objet, restreindre à 10 entrées max par émetteur
    df_export[["Libellé émetteur (ACTES)", "Objet"]].groupby(["Libellé émetteur (ACTES)"]).apply(lambda x: display(x.head(10)))

Libellé émetteur (ACTES)  \
71  3  ALLAUCH       MAIRIE   

                                                                                                                                                                                                                                                                                                        Objet  
71  Arrêté de mise en sécurité procédure urgente - mise en demeure d'effectuer sous 48 h : mise en place tous moyens d'étaiement de la sous-facee du plancher afin de sécuriser les occupants du seul logement constituant l'immeuble - M. et Mme BERGOT, Propriétaires, 6, rue des Michels 13190 ALLAUCH - ?

Libellé émetteur (ACTES)  \
0   3 AUBAGNE   MAIRIE        
62  3 AUBAGNE   MAIRIE        

                                                            Objet  
0   ARRETE MISE EN SECURITE PROCEDURE ORDINAIRE LE CLOS CHEVALIER  
62   ARRETE MAIN LEVEE DE TOUT PERIL 16 COURS FOCH

Libellé émetteur (ACTES)  \
1   3 LA CIOTAT     MAIRIE    
10  3 LA CIOTAT     MAIRIE    
56  3 LA CIOTAT     MAIRIE    
57  3 LA CIOTAT     MAIRIE    

                                                                                                          Objet  
1   Mise en sécurité de travaux d'urgence dans l'immeuble sis 14 rue M. Lieutaud                                 
10  Mise en sécurité de travaux d'urgence dans l'immeuble sis 9 rue Louis Vignol                                 
56  Mise en sécurité de travaux d'urgence dans l'immeuble sis 2 rue Ledru Rollin/8 rue Marius Lieutaud           
57  Mise en sécurité de travaux d'urgence dans l'immeuble sis 13 rue Camille Pelletan/ 4 rue des Frères Romana.

Libellé émetteur (ACTES)  \
2   3 MARSEILLE  MAIRIE       
3   3 MARSEILLE  MAIRIE       
4   3 MARSEILLE  MAIRIE       
5   3 MARSEILLE  MAIRIE       
6   3 MARSEILLE  MAIRIE       
7   3 MARSEILLE  MAIRIE       
8   3 MARSEILLE  MAIRIE       
9   3 MARSEILLE  MAIRIE       
11  3 MARSEILLE  MAIRIE       
12  3 MARSEILLE  MAIRIE       

                                                                                                                                                    Objet  
2   sdi 20/047 - arrêté modificatif de mise en sécurité - 7 rue séraphin - 13015 marseille - parcelle n°2015899 H0057                                      
3   SDI 22/037 - Arrêté de Mise en Sécurité - Procédure Urgente - traverse Marie Louise - 13015 MARSEILLE - Parcelle n°215905 O0186                        
4   SDI 19/031 - ARRÊTÉ MODIFICATIF DE MISE EN SÉCURITÉ - 20, RUE D'ANVERS - 13001 MARSEILLE - PARCELLE N°201805 D0220                                     
5   SDI 18/007 - Arrêté de Mise en Sécurité - 85, rue de la Palud - 13006 Marseille - Parcelle n°206827 A0202                                              
6   SDI 22/0071 - ARRÊTÉ MODIFICATIF DE L?ARRÊTÉ DE MISE EN SECURITE - PROCÉDURE URGENTE - 150 AVENUE DU MERLAN 13014 MARSEILLE - PARCELLE n°214893 B0163  
7   SDI  20/201 - ARRETE DE MISE EN SECURITE - PROCEDURE D'URGENCE - 8 RUE EUGÈNE POTTIER / 5 RUE HOCHE 13003 - PARCELLE N°203814 C0018                    
8   SDI 22/087 - Arrêté de mise en sécurité - Procédure urgente - ?2 rue Roger SCHIAFFINI - 13003 Marseille - Parcelle n°203 811 M0165                     
9   SDI 21/573 - ARRÊTÉ DE MAINLEVÉE D'ARRÊTÉ  DE MISE EN SÉCURITÉ - PROCÉDURE URGENTE - 19 RUE DE LA PALUD - 13001 MARSEILLE - PARCELLE N°201803 B0046    
11  SDI 22/109 - Arrêté de Mise en Sécurité - Procédure Urgente - 20 rue des Clairistes - 13005 MARSEILLE - Parcelle n°131205821 A0072                     
12  SDI 21/656 - Mise en sécurité - Procédure Urgente - 39-41 rue Condorcet 13016 MARSEILLE - Parcelle n°216910 E0188

Libellé émetteur (ACTES)  \
96  3 auriol mairie           

                                                                                                                                                                                                                                                                                           Objet  
96  ARRETE MODIFICATIF N.2022-02 DE MISE EN SECURITE - PROCEDURE URGENTE Immeuble 3 rue Salomon - Parcelle AC 137 (risques présentés par les murs, bâtiments ou édifices quelconques n'offrant pas les garanties de solidités nécessaires au maintien de la sécurité des occupants et des tiers)

Libellé émetteur (ACTES)  \
50  4 istres                  
63  4 istres                  

                                                                                                                                                                           Objet  
50  Arrêté de mise en sécurité - Procédure ordinaire. Immeuble cadastré section CK n.106 sis 12 rue Saint Jacques, propriété de Madame Tiffany Giove                              
63  Mainlevée de l'arrêté de mise en sécurité n.1783/2021 du 12 novembre 2021 - Immeuble cadastré section CK n.214 sis 3 place des Bourras - Propriété de M. Jean Pierre Coulomb


émetteur | donnée | champ @ctes | section du texte
:-|:-:|:-|:-
Allauch | date | N° Acte (*20220307*) | **erreur OCR** (tampon)
| numéro | 
| type | Objet (*mise en sécurité procédure urgente*) | titre (*MISE EN SECURITE PROCEDURE URGENTE*)
| adresse | **non** | article 1 (*18 Grand Rue à Allauch*)
| parcelle | **non** | article 1 (*EB 228*)
| 
Aubagne | type | oui (*MISE EN SECURITE PROCEDURE ORDINAIRE*, *MAIN LEVEE DE TOUT PERIL*) | 
| adresse | oui (*LE CLOS CHEVALIER*, *16 COURS FOCH*) | 
| parcelle | non | oui ()

NB: Pour Allauch, l'adresse dans le titre sur actes est celle des destinataires de l'arrêté (adresse des propriétaires).



## Matière

Deux "matières" ont été identifiées comme contenant des arrêtés de mise en sécurité:

* 6.1 - Police municipale
* 6.4 - Autres actes reglementaires

In [15]:
# TODO groupby("Matière") => vérifier si diff dans les autres colonnes (format champs texte, etc)
df_recher.groupby("Matière").describe()

N° Acte         \
                                    count unique   
Matière                                            
6.1 - Police municipale               108    108   
6.4 - Autres actes reglementaires       6      6   

                                                                           \
                                                                      top   
Matière                                                                     
6.1 - Police municipale               013-211300058-20220307-070322_01-AR   
6.4 - Autres actes reglementaires  013-211300868-20220210-AG46-2022BIS-AR   

                                       Nature                               \
                                  freq  count unique                   top   
Matière                                                                      
6.1 - Police municipale              1    108      1  Actes réglementaires   
6.4 - Autres actes reglementaires    1      6      1  Actes réglementaires   

                                       Objet         ... Arrondissement       \
                                  freq count unique  ...            top freq   
Matière                                              ...                       
6.1 - Police municipale            108   108    108  ...              3  108   
6.4 - Autres actes reglementaires    6     6      6  ...              3    6   

                                  Région                                     \
                                   count unique                         top   
Matière                                                                       
6.1 - Police municipale              108      1  Provence-Alpes-Côte-d'Azur   
6.4 - Autres actes reglementaires      6      1  Provence-Alpes-Côte-d'Azur   

                                       Strate         \
                                  freq  count unique   
Matière                                                
6.1 - Police municipale            108    108      1   
6.4 - Autres actes reglementaires    6      6      1   

                                                                                      \
                                                                                 top   
Matière                                                                                
6.1 - Police municipale            Population supérieure ou égale  à 1 000 000 ha...   
6.4 - Autres actes reglementaires  Population supérieure ou égale  à 1 000 000 ha...   

                                        
                                  freq  
Matière                                 
6.1 - Police municipale            108  
6.4 - Autres actes reglementaires    6  

[2 rows x 88 columns]